src/components/MediaViewer/HorizontalScroll.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import type { Node } from "react";
+import React, { useRef, useState } from "react";
+import { Dimensions, FlatList } from "react-native";
+
+import Photo from "../../models/Photo";
+import DeletePhotoDialog from "../SharedComponents/DeletePhotoDialog";
+import PhotoCarousel from "../SharedComponents/PhotoCarousel";
+import CustomImageZoom from "./CustomImageZoom";
+
+type Props = {
+  photoUris: Array<string>,
+  initialPhotoSelected: number,
+  deleteDialogVisible: boolean,
+  setPhotoUris: Function,
+  hideDialog: Function
+}
+
+const { width } = Dimensions.get( "screen" );
+
+const HorizontalScroll = ( {
+  photoUris, initialPhotoSelected, deleteDialogVisible, setPhotoUris, hideDialog
+}: Props ): Node => {
+  const horizontalScroll = useRef( null );
+  const [selectedPhotoIndex, setSelectedPhotoIndex] = useState( initialPhotoSelected );
+
+  const scrollToIndex = index => {
+    // when a user taps a photo in the carousel, the UI needs to automatically
+    // scroll to the index of the photo they selected
+    if ( !horizontalScroll?.current ) { return; }
+    setSelectedPhotoIndex( index );
+    horizontalScroll?.current.scrollToIndex( { index, animated: true } );
+  };
+
+  const renderImage = ( { item } ) => (
+    <CustomImageZoom source={{ uri: Photo.displayLargePhoto( item ) }} />
+  );
+
+  // need getItemLayout for setting initial scroll index
+  const getItemLayout = ( data, index ) => ( {
+    length: width,
+    offset: width * index,
+    index
+  } );
+
+  const handleScrollLeft = ( ) => {
+    if ( selectedPhotoIndex === 0 ) { return; }
+    setSelectedPhotoIndex( selectedPhotoIndex - 1 );
+  };
+
+  const handleScrollRight = ( ) => {
+    if ( selectedPhotoIndex === photoUris.length - 1 ) { return; }
+    setSelectedPhotoIndex( selectedPhotoIndex + 1 );
+  };
+
+  const handleScrollEndDrag = e => {
+    const { contentOffset } = e.nativeEvent;
+    const { x } = contentOffset;
+
+    const currentOffset = width * selectedPhotoIndex;
+
+    if ( x > currentOffset ) {
+      handleScrollRight( );
+    } else {
+      handleScrollLeft( );
+    }
+  };
+
+  return (
+    <>
+      <FlatList
+        data={photoUris}
+        renderItem={renderImage}
+        initialScrollIndex={selectedPhotoIndex}
+        getItemLayout={getItemLayout}
+        horizontal
+        pagingEnabled
+        showsHorizontalScrollIndicator={false}
+      // $FlowIgnore
+        ref={horizontalScroll}
+        onScrollEndDrag={handleScrollEndDrag}
```

Comment:
This approach is causing this bug: if you touch down, slide to the next image but don't release, then slide back to the original image, then release, this even seems to fire before the glide animation finishes, which means the `PhotoCarousel` highlights the next image while the `FlatList` is still showing the first.

I can sort of address this by using [`onMomentumScrollEnd`](https://reactnative.dev/docs/scrollview#onmomentumscrollend) instead ([thanks, SO](https://stackoverflow.com/questions/43370807/react-native-get-current-page-in-flatlist-when-using-pagingenabled)), but that seems to call the callback multiple times which causes jitter, and I'm not sure how to tell which event is the last without doing something unpleasant like `setTimeout`. Any ideas?